### PR TITLE
OldSindhi / MarathiCursive Fonts update

### PR DESCRIPTION
**Pull request by Mathnerd314**
###### Motivation for this change

Fallout from #86417 / #87837. The newer versions of the fonts use tar.xz and so don't need any special handling.

Or they could just be dropped. The original motivation for packaging these was Unicode coverage but Noto has added similar fonts that work fine.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested fonts show up in https://en.wiktionary.org/wiki/Appendix:Unicode/Modi and https://en.wikipedia.org/wiki/Khudawadi_(Unicode_block) correctly
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mathnerd314** [comment]
cc @flokli to potentially merge

File: pkgs/data/fonts/marathi-cursive/default.nix [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
+{ lib, fetchzip }:
 
 let
-  version = "1.2";
+  version = "2.0";
 in fetchzip rec {
   name = "marathi-cursive-${version}";
 
-  url = "https://github.com/MihailJP/MarathiCursive/releases/download/${version}/MarathiCursive-${version}.7z";
+  url = "https://github.com/MihailJP/MarathiCursive/releases/download/v2.0/MarathiCursive-2.0.tar.xz";
```
Comment (prusnak) [inline pkgs/data/fonts/marathi-cursive/default.nix:8]:
```suggestion
  url = "https://github.com/MihailJP/MarathiCursive/releases/download/v${version}/MarathiCursive-${version}.tar.xz";
```

File: pkgs/data/fonts/marathi-cursive/default.nix [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
 
   postFetch = ''
-    ${p7zip}/bin/7z x $downloadedFile
-    cd MarathiCursive
-
+    tar -xJf $downloadedFile --strip-components=1
     install -m444 -Dt $out/share/fonts/marathi-cursive *.otf *.ttf
-    install -m444 -Dt $out/share/doc/${name}           README *.txt
+    install -m444 -Dt $out/share/doc/${name}  README *.txt
```
Comment (prusnak) [inline pkgs/data/fonts/marathi-cursive/default.nix:13]:
```suggestion
    install -m444 -Dt $out/share/doc/${name} README *.txt
```

File: pkgs/data/fonts/oldsindhi/default.nix [hunk 3]
```diff
@@ -1,26 +1,25 @@
...
+{ lib, fetchzip }:
 
 let
-  version = "0.1";
+  version = "1.0";
 in fetchzip rec {
   name = "oldsindhi-${version}";
 
-  url = "https://github.com/MihailJP/oldsindhi/releases/download/0.1/OldSindhi-0.1.7z";
+  url = "https://github.com/MihailJP/oldsindhi/releases/download/v1.0/OldSindhi-1.0.tar.xz";
```
Comment (prusnak) [inline pkgs/data/fonts/oldsindhi/default.nix:8]:
```suggestion
  url = "https://github.com/MihailJP/oldsindhi/releases/download/v${version}/OldSindhi-${version}.tar.xz";
```

File: pkgs/data/fonts/oldsindhi/default.nix [hunk 4]
```diff
@@ -1,26 +1,25 @@
...
 
   postFetch = ''
-    ${p7zip}/bin/7z x $downloadedFile
-
-    install -m444 -Dt $out/share/fonts/truetype OldSindhi/*.ttf
-    install -m444 -Dt $out/share/doc/${name}    OldSindhi/README OldSindhi/*.txt
+    tar -xJf $downloadedFile --strip-components=1
+    install -m444 -Dt $out/share/fonts/truetype *.ttf
+    install -m444 -Dt $out/share/doc/${name}    README *.txt
```
Comment (prusnak) [inline pkgs/data/fonts/oldsindhi/default.nix:13]:
```suggestion
    install -m444 -Dt $out/share/doc/${name} README *.txt
```

**Review by prusnak — CHANGES_REQUESTED** [review]

**Review by prusnak — APPROVED** [review]

**Comment by flokli** [comment]
Can the style nits be squashed into the corresponding earlier commits?

**Comment by Mathnerd314** [comment]
@flokli Done.

**Comment by flokli** [comment]
Thanks!

**Review by prusnak — APPROVED** [review]

**Pull request merged**